mw-webapp/src/pages/authPage/AuthPage.tsx

```diff
@@ -1,14 +1,62 @@
-import {Link} from "react-router-dom";
+import {User, getRedirectResult, onAuthStateChanged} from "firebase/auth";
+import {useEffect, useState} from "react";
+import {Link, useNavigate} from "react-router-dom";
 import {useAuth} from "src/utils/useAuth";
 
 export const AuthPage = () => {
-  const handleGoogleSignIn = useAuth();
+  const {handleGoogleSignIn, handleLogout, writeNewUserData, auth} = useAuth();
+  const navigate = useNavigate();
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    const handleAuthState = onAuthStateChanged(auth, (currentUser) => {
+      if (currentUser) {
+        setUser(currentUser);
+      } else {
+        setUser(null);
+      }
+    });
+    return () => {
+      handleAuthState();
+    };
+  }, []);
+
+  useEffect(() => {
+    const getNewUserCredentials = async () => {
+      try {
+        const userCredentials = await getRedirectResult(auth);
+        if (userCredentials) {
+          writeNewUserData(
+            userCredentials.user.uid, userCredentials.user.email, userCredentials.user.displayName,
+          );
+          navigate("/main");
+        } else {
+          return;
+        }
+      } catch (error) {
+        let errorMessage;
+        if (error instanceof Error) {
+          errorMessage = error.message;
+        }
+        alert(errorMessage);
+      }
+    };
+    return () => {
+      getNewUserCredentials();
+    };
+  }, []);
 
```

Comment:
Again if possible bette rmove out function. And call it inside useEffect. Also no eed to use 2 useEffect. You can call few functions in 1 useEffect

Reply:
Done